source/Halibut.Tests/Transport/ConnectionManagerExtensionMethods.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Halibut.Diagnostics;
+using Halibut.Tests.Support.TestAttributes;
+using Halibut.Transport;
+using Halibut.Transport.Protocol;
+
+namespace Halibut.Tests.Transport
+{
+    public static class ConnectionManagerExtensionMethods
+    {
+        public static async Task<IConnection> AcquireConnection_SyncOrAsync(this ConnectionManager connectionManager, SyncOrAsync syncOrAsync, ExchangeProtocolBuilder exchangeProtocolBuilder, IConnectionFactory connectionFactory, ServiceEndPoint serviceEndPoint, ILog log, CancellationToken cancellationToken)
+        {
+#pragma warning disable CS0612 // Type or member is obsolete
+            return await syncOrAsync
+                .WhenSync(() => connectionManager.AcquireConnection(exchangeProtocolBuilder, connectionFactory, serviceEndPoint, log, cancellationToken))
+                .WhenAsync(async () => await connectionManager.AcquireConnectionAsync(exchangeProtocolBuilder, connectionFactory, serviceEndPoint, log, cancellationToken));
+#pragma warning restore CS0612 // Type or member is obsolete
```

Comment:
This slightly breaks our don't edit any sync code path, however if this didn't work it would fail almost every test.

Reply:
wait no never mind, facepalm this is in test and exists for the test below. TODO drink more yerba mate.